src/Nethermind/Nethermind.JsonRpc/Modules/Eth/EthModule.TransactionExecutor.cs

```diff
@@ -0,0 +1,180 @@
+        {
+            public EstimateGasTransactionExecutor(IBlockchainBridge blockchainBridge, IBlockFinder blockFinder, TimeSpan cancellationTokenTimeout, IJsonRpcConfig rpcConfig) 
+                : base(blockchainBridge, blockFinder, cancellationTokenTimeout, rpcConfig)
+            {
+            }
+
+            protected override BlockchainBridge.CallOutput ExecuteTx(BlockHeader header, Transaction tx, CancellationToken token) => 
+                _blockchainBridge.EstimateGas(header, tx, token);
+
+            protected override UInt256? GetResult(Transaction tx, BlockchainBridge.CallOutput result) => (UInt256)result.GasSpent;
+
+            protected override ResultWrapper<UInt256?> GetExecutionError(Transaction tx, BlockchainBridge.CallOutput result) => 
+                ResultWrapper<UInt256?>.Fail(result.Error, ErrorCodes.InternalError);
+        }
+        
+        private class CreateAccessListTransactionExecutor : TransactionExecutor<AccessListForRpc>
+        {
+            private readonly bool _optimize;
+
+            public CreateAccessListTransactionExecutor(IBlockchainBridge blockchainBridge, IBlockFinder blockFinder, TimeSpan cancellationTokenTimeout, IJsonRpcConfig rpcConfig, bool optimize) 
+                : base(blockchainBridge, blockFinder, cancellationTokenTimeout, rpcConfig)
+            {
+                _optimize = optimize;
+            }
+
+            protected override BlockchainBridge.CallOutput ExecuteTx(BlockHeader header, Transaction tx, CancellationToken token) => 
+                _blockchainBridge.CreateAccessList(header, tx, token, _optimize);
+
+            protected override AccessListForRpc GetResult(Transaction tx, BlockchainBridge.CallOutput result) =>
+                new(GetResultAccessList(tx, result), GetResultGas(tx, result));
+            
+
+            protected override ResultWrapper<AccessListForRpc> GetExecutionError(Transaction tx, BlockchainBridge.CallOutput result) => 
+                ResultWrapper<AccessListForRpc>.Fail(result.Error, ErrorCodes.InternalError, new(GetResultAccessList(tx, result), GetResultGas(tx, result)));
+            
+            private static AccessListItemForRpc[] GetResultAccessList(Transaction tx, BlockchainBridge.CallOutput result)
+            {
+                AccessList? accessList = result.AccessList ?? tx.AccessList;
+                return accessList is null ? Array.Empty<AccessListItemForRpc>() : AccessListItemForRpc.FromAccessList(accessList);
+            }
+            
+            private static UInt256 GetResultGas(Transaction transaction, BlockchainBridge.CallOutput result)
+            {
+                long gas = result.GasSpent;
+                if (result.AccessList is not null)
+                {
+                    // if we generated access list, we need to fix actual gas cost, as all storage was considered warm 
+                    gas -= IntrinsicGasCalculator.Calculate(transaction, Berlin.Instance);
```

Comment:
strange

Reply:
Its because user can already specify parts of AccessList on the input and we don't want to price it twice